examples/typescript/swap.ts

```diff
@@ -0,0 +1,247 @@
+/**
+ * Example to demonstrate creating and adding to liquidity pools, swapping between two fungible asset.
+ *
+ * Steps:
+ * 1. Create two accounts (Alice and Bob)
+ * 2. Fund Alice and Bob
+ * 3. Create a FA with Alice as the owner
+ * 4. Create another FA with Bob as the owner
+ * 5. Create a liquidity pool with Alice's FA and Bob's FA
+ * 6. Swap between Alice's FA and Bob's FA
+ */
+
+import { Account, AccountAddress, Aptos, Bool, Ed25519PrivateKey, U64, ViewRequestData } from "aptos";
+
+const readline = require("readline").createInterface({
+  input: process.stdin,
+  output: process.stdout,
+});
+
+const getOptimalLpAmount = async (
+  aptos: Aptos,
+  swap: AccountAddress,
+  token1Addr: AccountAddress,
+  token2Addr: AccountAddress,
+): Promise<void> => {
+  const payload: ViewRequestData = {
+    function: `0x${swap.toStringWithoutPrefix()}::router::optimal_liquidity_amounts`,
+    functionArguments: [token1Addr.toString(), token2Addr.toString(), false, "200000", "300000", "200", "300"],
+  };
+  const result = await aptos.view({ payload });
+  console.log("Optimal LP amount: ", result);
+};
+
+const addLiquidity = async (
+  aptos: Aptos,
+  swap: AccountAddress,
+  deployer: Account,
+  token1Addr: AccountAddress,
+  token2Addr: AccountAddress,
+): Promise<string> => {
+  const rawTxn = await aptos.generateTransaction({
+    sender: deployer.accountAddress.toString(),
+    data: {
+      function: `0x${swap.toStringLongWithoutPrefix()}::router::add_liquidity_entry`,
+      functionArguments: [
+        token1Addr,
+        token2Addr,
+        new Bool(false),
+        new U64(200000),
+        new U64(300000),
+        new U64(200),
+        new U64(300),
+      ],
+    },
+  });
+  const pendingTxn = await aptos.signAndSubmitTransaction({ signer: deployer, transaction: rawTxn });
+  const response = await aptos.waitForTransaction({ transactionHash: pendingTxn.hash });
+  console.log("Add liquidity succeed. - ", response.hash);
+  return response.hash;
+};
+
+const swap = async (
+  aptos: Aptos,
+  swap: AccountAddress,
+  deployer: Account,
+  fromToken: AccountAddress,
+  toToken: AccountAddress,
+  amountIn: number,
+  amountOutMin: number,
+  recipient: AccountAddress,
+): Promise<string> => {
+  const rawTxn = await aptos.generateTransaction({
+    sender: deployer.accountAddress.toString(),
```

Comment:
Is this required even with the signAndSubmitTransaction call on line 79?

Reply:
Yep this is required. I think this is basically generating raw txn in bytecode, and we need the address to specify where the module gonna live in.